charts/druid/values.yaml

```diff
@@ -1,13 +1,56 @@
-crds:
-  enabled: true
 image:
   repository: europe-docker.pkg.dev/gardener-project/public/gardener/etcd-druid
   tag: latest
   imagePullPolicy: IfNotPresent
 replicas: 1
-ignoreOperationAnnotation: false
-# enableBackupCompaction: true
-# eventsThreshold: 15
-# metricsScrapeWaitDuration: "10s"
-# featureGates:
-#   UseEtcdWrapper: true
+resources:
+  limits:
+    cpu: 300m
+    memory: 512Mi
+  requests:
+    cpu: 50m
+    memory: 128Mi
+
+featureGates:
+  UseEtcdWrapper: true
+
+controllerManager:
+  server:
+    metrics:
+      bindAddress: ""
+      port: 8080
+    webhook:
+      bindAddress: ""
+      port: 9443
+      tls:
+        serverCertDir: /etc/webhook-server-tls
+  leaderElection:
+    enabled: true
+    id: druid-leader-election
+  disableLeaseCache: true
+
+controllers:
+  etcd:
+    workers: 3
+    enableEtcdSpecAutoReconcile: false
+    disableEtcdServiceAccountAutomount: false
+    etcdStatusSyncPeriod: 15s
+    etcdMemberNotReadyThreshold: 5m
+    etcdMemberUnknownThreshold: 1m
+  compaction:
+    enabled: true
+    workers: 3
+    etcdEventsThreshold: 1000000
+    activeDeadlineDuration: 3h
+    metricsScrapeWaitDuration: 0s
+  etcdCopyBackupsTask:
+    workers: 3
+  secret:
+    workers: 10
+
+webhooks:
+  sentinel:
+    enabled: true
```

Comment:
In a previous review comment, it was mentioned that the sentinel webhook would be opt-in for consumers of etcd-druid, but the default behavior is that it is already opted in for the consumer. Should this be false? Or is this set to true for now to help reviewers while running etcd-druid?

When deploying etcd-druid locally through localstack, by following the steps in `getting-started-locally-localstack.md`; I see the following logs:

```
{"level":"info","ts":"2024-05-08T13:34:16.443Z","msg":"Starting EventSource","controller":"compaction-controller","controllerGroup":"druid.gardener.cloud","controllerKind":"Etcd","source":"kind source: *v1.Job"}
{"level":"info","ts":"2024-05-08T15:39:26.767Z","logger":"etcd-controller","msg":"Member hasn't acquired lease yet, still in bootstrapping phase","etcd":{"name":"etcd-test","namespace":"default"},"operation":"reconcileStatus","runID":"fd04811e-cceb-4ced-974a-3b49e69a6991","name":"etcd-test-0"}
{"level":"info","ts":"2024-05-08T15:39:41.785Z","logger":"etcd-controller","msg":"Member hasn't acquired lease yet, still in bootstrapping phase","etcd":{"name":"etcd-test","namespace":"default"},"operation":"reconcileStatus","runID":"d4facbe6-dcd2-41e8-bdfd-d3d432ac2394","name":"etcd-test-0"}
{"level":"info","ts":"2024-05-08T15:39:44.273Z","logger":"compaction-lease-controller","msg":"Compaction job reconciliation started"}
{"level":"info","ts":"2024-05-08T15:39:44.273Z","logger":"compaction-lease-controller","msg":"No compaction job currently running","etcd":"default/etcd-test","namespace":"default"}
{"level":"info","ts":"2024-05-08T15:41:15.219Z","logger":"secret-controller","msg":"Removing finalizer","secret":{"name":"etcd-backup-aws","namespace":"default"},"namespace":"default","name":"etcd-backup-aws","finalizerName":"druid.gardener.cloud/etcd-druid"}
{"level":"info","ts":"2024-05-08T15:41:15.250Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test","namespace":"default","resourceGroupKind":"/ServiceAccount","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.251Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test-peer","namespace":"default","resourceGroupKind":"/Service","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.314Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test-0","namespace":"default","resourceGroupKind":"coordination.k8s.io/Lease","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.336Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test-delta-snap","namespace":"default","resourceGroupKind":"coordination.k8s.io/Lease","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.336Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test-client","namespace":"default","resourceGroupKind":"/Service","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.339Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"druid.gardener.cloud:etcd:etcd-test","namespace":"default","resourceGroupKind":"rbac.authorization.k8s.io/RoleBinding","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.339Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-bootstrap-b8b642","namespace":"default","resourceGroupKind":"/ConfigMap","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.343Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"druid.gardener.cloud:etcd:etcd-test","namespace":"default","resourceGroupKind":"rbac.authorization.k8s.io/Role","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
2024/05/08 15:41:15 http: TLS handshake error from 10.244.0.1:43576: EOF
{"level":"info","ts":"2024-05-08T15:41:15.343Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test","namespace":"default","resourceGroupKind":"apps/StatefulSet","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.343Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test","namespace":"default","resourceGroupKind":"policy/PodDisruptionBudget","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
{"level":"info","ts":"2024-05-08T15:41:15.344Z","logger":"sentinel-webhook","msg":"Sentinel webhook invoked","name":"etcd-test-full-snap","namespace":"default","resourceGroupKind":"coordination.k8s.io/Lease","operation":"DELETE","user":"system:serviceaccount:kube-system:generic-garbage-collector"}
2024/05/08 15:41:15 http: TLS handshake error from 10.244.0.1:55982: EOF
2024/05/08 15:41:15 http: TLS handshake error from 10.244.0.1:45431: EOF
2024/05/08 15:41:15 http: TLS handshake error from 10.244.0.1:21136: EOF
```

Could you run with localstack once and see if any changes are needed to make it work with localstack as well?

Reply:
> In a previous review comment, it was mentioned that the sentinel webhook would be opt-in for consumers of etcd-druid, but the default behavior is that it is already opted in for the consumer. Should this be false? 

No, this is intended. For users who run druid locally as a process outside of a k8s environment, this feature should not be enabled, because making a webhook work outside of a cluster is a nightmare, and moreover unneeded. Whereas, when a user wants to run druid within a k8s cluster, by deploying it via our helm charts, then enabling the webhook makes sense, since that's where it's really needed.

But yes, you're right in saying that webhook should be disabled by default, since it's a new feature that users should opt in for. I will set it to `false` in the helm chart, but enable it for our pipelines/tests via the skaffold configuration, which is used when running `make deploy` and `make test-e2e`.

Reply:
I am unsure about the `http: TLS handshake error from <>: EOF` error messages, because I see from the logs that the webhook is actually working and processing requests.

I see that the community is facing similar errors too, with two ongoing issues open from a long time:
https://github.com/golang/go/issues/50984
https://github.com/kubernetes/kubernetes/issues/109022

This might explain these logs. Even if the connection is lost/dropped when serving the admission request, since the `failurePolicy` is set to `Fail` in the validating webhook config, the the request will automatically be denied, so the update/delete operation may fail, in which case the etcd reconciler simply records the error and requeues the operation, and the next time it should succeed.

Reply:
Thanks for the explanation.
This comment is exactly what we're seeing as well. https://github.com/kubernetes/kubernetes/issues/109022#issuecomment-1591260318.
Will keep a note of this.